store/views.py

```diff
@@ -86,21 +111,78 @@ def add_to_bag(request) -> JsonResponse:
     if not isinstance(product_id, int):
         raise TypeError("Product ID must be an integer.")
 
-    bag: list = request.session.get('bag', [])
+    bag: list = request.session.get("bag", [])
 
     product_found = False
     for product in bag:
-        if product['product_id'] == product_id:
-            product['quantity'] += 1
+        if product["product_id"] == product_id:
+            product["quantity"] += 1
             product_found = True
             break
-    
+
     if not product_found:
-        bag.append({'product_id': product_id, 'quantity': 1})
+        bag.append({"product_id": product_id, "quantity": 1})
+
+    request.session["bag"] = bag
+
+    total_items = sum(product["quantity"] for product in bag)
+    request.session["total_items"] = total_items
 
-    request.session['bag'] = bag
+    return JsonResponse({"status": "success", "total_items": total_items})
 
-    total_items = sum(product['quantity'] for product in bag)
-    request.session['total_items'] = total_items
 
-    return JsonResponse({'status': 'success', 'total_items': total_items})
\ No newline at end of file
+class ProductAdmin(LoginRequiredMixin, ListView):
+    model = Product
+    context_object_name = "products"
+    template_name = "store/product_admin.html"
+    login_url = "/accounts/login"
+
+    def get_queryset(self) -> QuerySet[Product]:
+        # return products only from the store they own
+        return Product.objects.filter(store__owner=self.request.user)
+
+    def get_context_data(self, **kwargs: Any) -> dict[str, Any]:
+        context = super().get_context_data(**kwargs)
+
+        context["store"] = self.request.user.store_set.get()
+        return context
+
+
+@login_required
+def product_admin_modify(request, pk):
+    product = get_object_or_404(Product, pk=pk)
```

Comment:
Hmm, here I would also check that `store__owner=request.user` or include a conditional such as: 
```
if product.store.owner != request.user:
        return HttpResponseForbidden("You do not have permission to modify this product.")
```
- But I believe the first way is more ideal and reduces additional db queries.